kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -9211,7 +9211,11 @@ protected Map fixTypeAndId(String id, String type)
 		extType.put("id", id);
 		if (type == null) type = "";
 		extType.put("type", type);
-		String extension = Validator.getFileExtension(id);
+		
+		String[] idSegments;
+		idSegments = id.split("/");
+		String filename = idSegments.length > 0 ? idSegments[idSegments.length - 1]:id;
+		String extension = Validator.getFileExtension(filename);
```

Comment:
I do think you're right about having utility methods to handle was is in effect entity parsing so that code at a high level and so easier to read a less bug prone.